### PR TITLE
Fixed focus management.

### PR DESCRIPTION
**Pull request by GioBonvi**
Rewrote the `_Focus()` function to give focus to the foremost window in the desktop.

Replaced the `_FocusIfRequested()` function with `_Focus()`.

Had to update the library to the latest version as the `IsWindowOnDesktopNumber` function seemed not to work in the version previoulsy used.

This solves #64 and possibly even the glitch of the flashing orange icons in the taskbar: I could not reproduce it in my tests for this version of the code, while with the stable release it happened quite often.

### CODE REVIEW

File: virtual-desktop-enhancer.ahk [hunk 1]
```diff
@@ -535,16 +534,30 @@ _ChangeAppearance(n:=1) {
...
+_Focus() {
+    foremostWindowId := _GetforemostWindowIdOnDesktop(_GetCurrentDesktopNumber())
+    WinActivate, ahk_id %foremostWindowId%
 }
 
-_Focus() {
-    WinActivate, ahk_class Shell_TrayWnd
-    SendEvent !{Esc}
+; Select the ahk_id of the foremost window in a given virtual desktop.
+_GetforemostWindowIdOnDesktop(n) {
```
Comment (sdias) [inline virtual-desktop-enhancer.ahk]: `GetforemostWindowIdOnDesktop` to `GetForemostWindowIdOnDesktop`

File: virtual-desktop-enhancer.ahk [hunk 2]
```diff
@@ -535,16 +534,30 @@ _ChangeAppearance(n:=1) {
     }
 }
 
-_FocusIfRequested() {
```
Comment (sdias) [inline virtual-desktop-enhancer.ahk]: Does removing this not reintroduce #11?

Comment (GioBonvi) [inline virtual-desktop-enhancer.ahk]: Yes, it does: fixing it now.

**Comment by sdias** [comment]
Good work all around, just a few comments. :)

**Comment by sdias** [comment]
It feels a lot more responsive, great work, the solution is pretty elegant too.
I did notice a bit of an annoying behavior though. If I'm playing a game in fullscreen windowed mode, when I switch to that desktop it focus on the game, which then grabs the cursor and places it in the center of the screen, but since the tooltip for the desktop is still there, if you try clicking you actually end up clicking the tooltip, so you kind of have to wait for the tooltip to disappear.
I think we should make it so that clicks on the tooltip pass through it and click whatever window is below it instead. I know that currently, we use a click on the tooltip to dismiss it, but it is shown for so little time that that use case is not very useful in my opinion. If users don't agree, we can make it configurable. What do you think?

**Comment by GioBonvi** [comment]
You are right, probably noone has ever used the dismiss-on-click function: popups are just too fast; making popups click-through is the best solution for me.

**Comment by sdias** [comment]
Maybe do it in a separate PR to make it easier to review. Thanks.

**Comment by GioBonvi** [comment]
Ok, doing it now.

Can this be merged?

**Comment by sdias** [comment]
Yes, I'll merge it. Going forward I'll do proper reviews so you can know when I fully reviewed it and approved it, so that then you can merge it when you want.

**Pull request merged**